apps/src/storage/dataBrowser/DataVisualizer.jsx

```diff
@@ -0,0 +1,135 @@
+    // https://firebase.googleblog.com/2014/04/best-practices-arrays-in-firebase.html
+    tableRecords: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
+      .isRequired
+  };
+
+  state = {...INITIAL_STATE};
+
+  handleClose = () => {
+    this.setState({isVisualizerOpen: false});
+  };
+
+  handleChange = (field, value) => {
+    this.setState({[field]: value});
+  };
+
+  render() {
+    const dropdownField = (fieldName, displayName, values) => (
+      <div id={fieldName} style={rowStyle.container}>
+        <label style={rowStyle.description}>{displayName}</label>
+        <select
+          value={this.state[fieldName]}
+          onChange={event => this.handleChange(fieldName, event.target.value)}
+        >
+          <option value="">Select</option>
+          {values.map(value => (
+            <option key={value} value={value}>
+              {value}
+            </option>
+          ))}
+        </select>
+      </div>
+    );
+
+    const modalBody = (
+      <div>
+        <h1> Explore {this.props.tableName} </h1>
+        <h2> Overview </h2>
+        <div id="selection-area">
+          <div id="chartTitleRow" style={rowStyle.container}>
+            <label style={rowStyle.description}>Chart Title</label>
+            <input
+              style={rowStyle.input}
+              onChange={event =>
+                this.handleChange('chartTitle', event.target.value)
+              }
+            />
+          </div>
+
+          {dropdownField('chartType', 'Chart Type', [
+            'Bar Chart',
+            'Histogram',
+            'Cross Tab',
+            'Scatter Plot'
+          ])}
+
+          {this.state.chartType === 'Histogram' && (
+            <div id="numBinsRow" style={rowStyle.container}>
+              <label style={rowStyle.description}>Bins</label>
+              <input
+                style={rowStyle.input}
+                value={this.state.numBins}
+                onChange={event =>
+                  this.handleChange('numBins', event.target.value)
+                }
+              />
+            </div>
+          )}
+          {(this.state.chartType === 'Bar Chart' ||
+            this.state.chartType === 'Histogram') &&
+            dropdownField('values', 'Values', this.props.tableColumns)}
+
+          {(this.state.chartType === 'Cross Tab' ||
+            this.state.chartType === 'Scatter Plot') && (
+            <div>
+              {dropdownField('xValues', 'X Values', this.props.tableColumns)}
+              {dropdownField('yValues', 'Y Values', this.props.tableColumns)}
+            </div>
+          )}
+        </div>
+        <div id="chart-area" />
+      </div>
+    );
+
+    return (
+      <span style={[{display: 'inline-block'}]}>
+        <button
+          type="button"
+          style={dataStyles.whiteButton}
+          onClick={() => this.setState({isVisualizerOpen: true})}
+        >
+          Show Viz (Placeholder)
+        </button>
+        <BaseDialog
```

Comment:
@epeach Did we officially decide on a dialog to use everywhere? Dialog? BaseDialog? LegacyDialog (Ok, I was joking about that last one).

Reply:
(Doesn't need to block this PR)

Reply:
I ended up using Dialog so I had access to the Title/Footer/Body stuff. What did you end up using?